xarray/core/missing.py

```diff
@@ -332,3 +369,162 @@ def _get_valid_fill_mask(arr, dim, limit):
     return (arr.isnull().rolling(min_periods=1, **kw)
             .construct(new_dim, fill_value=False)
             .sum(new_dim, skipna=False)) <= limit
+
+
+def _assert_single_chunk(obj, axes):
+    for axis in axes:
+        if len(obj.chunks[axis]) > 1 or obj.chunks[axis][0] < obj.shape[axis]:
+            raise ValueError('Chunking along the dimension to be interpolated '
+                             '({}) is not allowed.'.format(axis))
+
+
+def _localize(obj, indexes_coords):
+    """ Speed up for linear and nearest neighbor method.
+    Only consider a subspace that is needed for the interpolation
+    """
+    indexes = {}
+    for dim, [x, new_x] in indexes_coords.items():
+        index = x.to_index()
+        imin = index.get_loc(np.min(new_x.values), method='nearest')
+        imax = index.get_loc(np.max(new_x.values), method='nearest')
+
+        indexes[dim] = slice(max(imin - 2, 0), imax + 2)
+        indexes_coords[dim] = (x[indexes[dim]], new_x)
+    return obj.isel(**indexes), indexes_coords
+
+
+def interp(obj, indexes_coords, method, **kwargs):
+    """ Make an interpolation of Variable
+
+    Parameters
+    ----------
+    obj: Variable
+    index_coord:
+        mapping from dimension name to a pair of original and new coordinates.
+        Original coordinates should be sorted in strictly ascending order.
+    method: string
+        One of {'linear', 'nearest', 'zero', 'slinear', 'quadratic',
+        'cubic'}. For multidimensional interpolation, only
+        {'linear', 'nearest'} can be used.
+    kwargs:
+        keyword arguments to be passed to scipy.interpolate
+
+    Returns
+    -------
+    Interpolated Variable
+    """
+    if not indexes_coords:
+        return obj.copy()
+
+    # simple speed up for the local interpolation
+    if method in ['linear', 'nearest']:
+        obj, indexes_coords = _localize(obj, indexes_coords)
+
+    if method not in ['linear', 'nearest', 'zero', 'slinear', 'quadratic',
+                      'cubic', 'polynomial']:
+        raise ValueError('{} is not a valid method.'.format(method))
+
+    # default behavior
+    kwargs['bounds_error'] = kwargs.get('bounds_error', False)
+    # as coords should be always sorted, neglect this option
+    kwargs.pop('assume_sorted', None)
```

Comment:
I'm wondering if this check is actually needed. I'm not seeing where this kwarg is coming from.

Reply:
Agreed. removed.